privacyidea/lib/tokenclass.py

```diff
@@ -449,6 +451,59 @@ def authenticate(self, passw, user=None, options=None):
 
         return pin_match, otp_counter, reply
 
+    @staticmethod
+    def decode_base32check(encoded_otpkey):
+        """
+        Decode the OTP key which is given in the following format:
+
+            strip_padding(base32(sha1(otpkey)[:4] + otpkey))
+
+        Raise a ParameterError if the OTP key is malformed.
+        :return: hex-encoded otpkey
+        """
+        # First, add the padding to have a multiple of 8 bytes
+        encoded_length = len(encoded_otpkey)
+        if encoded_length % 8 != 0:
+            encoded_otpkey += "=" * (8 - (encoded_length % 8))
+        assert len(encoded_otpkey) % 8 == 0
+        # Decode as base32
+        try:
+            decoded_otpkey = base64.b32decode(encoded_otpkey)
+        except TypeError:
+            raise ParameterError("Malformed base32check OTP key: Invalid base32")
+        # Extract checksum and otpkey
+        if len(decoded_otpkey) < 4:
+            raise ParameterError("Malformed base32check OTP key: Too short")
+        checksum, otpkey = decoded_otpkey[:4], decoded_otpkey[4:]
+        otpkey_hash = hashlib.sha1(otpkey).digest()
+        if otpkey_hash[:4] != checksum:
+            raise ParameterError("Malformed base32check OTP key: Incorrect checksum")
+        return binascii.hexlify(otpkey)
+
+    @staticmethod
+    def decode_otpkey(otpkey, otpkeyformat):
```

Comment:
This function is used to decode the OTP from the API parameters. So this naming is fine.

I know, I am nitpicking.